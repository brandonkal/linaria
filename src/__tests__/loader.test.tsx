import pack from './__utils__/compiler';
import rimraf from 'rimraf';
import Module from '../babel/module';
import path from 'path';

describe('webpack loader', () => {
  beforeEach(() => {
    rimraf.sync(path.join(__dirname, './__fixtures__/.linaria-cache'));
    Module.invalidateAll();
  });
  test('loader requires generated CSS file - css', async () => {
    const FILE = 'css-classname.js';
    const stats = await pack({ fixture: FILE });
    const modules: any[] = stats.toJson().modules;
    const output = modules.find(v => v.name.endsWith(FILE)).source;

    expect(output).toMatchSnapshot();
  });

  test('loader requires generated CSS file - styled', async () => {
    const FILE = 'styled-components.js';
    const stats = await pack({ fixture: FILE });
    const modules: any[] = stats.toJson().modules;
    const output = modules.find(v => v.name.endsWith(FILE)).source;

    expect(output).toMatchSnapshot();
  });

  test('complex loader works -- no-optimize', async () => {
    const FILE = 'styled-components.js';
    const stats = await pack({
      fixture: FILE,
      production: false,
      complex: true,
    });
    const modules: any[] = stats.toJson().modules;
    const output = modules.find(v => v.name.endsWith(FILE)).source;
    const css: string = stats.compilation.assets['styles.css'].source();
    expect(css.includes('LINARIA')).toBe(false);

    expect(output).toMatchSnapshot();
  });

  test('complex loader works -- optimizer', async () => {
    const FILE = 'styled-components.js';
    const stats = await pack({
      fixture: FILE,
      production: true,
      complex: true,
      optimize: true,
    });
    const modules: any[] = stats.toJson().modules;
    let js = modules.find(v => v.name.includes(FILE));
    if (js.modules && js.modules.length) {
      js = js.modules.find(v => v.name.includes(FILE));
    }
    const jsSrc = js.source;
    const bundle: string = stats.compilation.assets['bundle.js'].source();
    const css: string = stats.compilation.assets['styles.css'].source();
    expect(css.includes('LINARIA')).toBe(false);
    expect(bundle.includes('LINARIA')).toBe(false);
    const snap =
      '// JS after Babel plugin before optimization\n' +
      jsSrc +
      '\n\n/* CSS after optimization */\n' +
      css;

    expect(snap).toMatchSnapshot();
  });
});
