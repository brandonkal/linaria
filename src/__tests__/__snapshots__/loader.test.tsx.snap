// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack loader complex loader works -- no-optimize 1`] = `
"import { styled } from '@brandonkal/linaria/react';
export var T1 =
/*#__PURE__*/
styled(\\"h1\\")({
  name: \\"T1\\",
  class: \\"T1_tjd0u1n\\",
  vars: {
    \\"tjd0u1n-0\\": [() => window.innerWidth * 0.5, \\"px\\"]
  }
});
export var T2 =
/*#__PURE__*/
styled(\\"h2\\")({
  name: \\"T2\\",
  class: \\"T2_t1jdyzzg\\",
  mod: {
    \\"t1jdyzzg__primary_0\\": p => p.primary$
  }
});
export var T3 =
/*#__PURE__*/
styled(\\"h3\\")({
  name: \\"T3\\",
  class: \\"T3_t1y4fq36\\"
}); // Injected by Linaria Loader for Webpack Compilation (virtual)

require(\\"./.linaria-cache/css/d8444e21c14a3c7e7c923bbcf86d4d84.dev.linaria.css\\");"
`;

exports[`webpack loader complex loader works -- optimizer 1`] = `
"// JS after Babel plugin before optimization
import { styled } from '@brandonkal/linaria/react';
export var T1 =
/*#__PURE__*/
styled(\\"h1\\")({
  name: \\"T1\\",
  class: \\"LINARIA_tjd0u1n_LINARIA\\",
  vars: {
    \\"LINARIA_tjd0u1n_LINARIA-0\\": [() => window.innerWidth * 0.5, \\"px\\"]
  }
});
export var T2 =
/*#__PURE__*/
styled(\\"h2\\")({
  name: \\"T2\\",
  class: \\"LINARIA_t1jdyzzg_LINARIA\\",
  mod: {
    \\"LINARIA_t1jdyzzg_LINARIA-0\\": p => p.primary$
  }
});
export var T3 =
/*#__PURE__*/
styled(\\"h3\\")({
  name: \\"T3\\",
  class: \\"LINARIA_t1y4fq36_LINARIA\\"
}); // Injected by Linaria Loader for Webpack Compilation (virtual)

require(\\"./.linaria-cache/css/d8444e21c14a3c7e7c923bbcf86d4d84.prod.linaria.css\\");

/* CSS after optimization */


.c {
  background: #111;
  width: var(--c-0);
}
.a {
  background: #222;
}
.a.a-0 {
    color: red;
  }
.b .T2_t1jdyzzg {
    background: #333;
  }
"
`;

exports[`webpack loader loader requires generated CSS file - css 1`] = `
"import { css } from '@brandonkal/linaria';
const Container = \\"Container_c1j2k3d2\\";
export default Container;

// Injected by Linaria Loader for Webpack Compilation (virtual)
require(\\"./.linaria-cache/css/cff3476c374622df2ed1295f40efaa97.dev.linaria.css\\");"
`;

exports[`webpack loader loader requires generated CSS file - styled 1`] = `
"import { styled } from '@brandonkal/linaria/react';
export const T1 =
/*#__PURE__*/
styled(\\"h1\\")({
  name: \\"T1\\",
  class: \\"T1_tjd0u1n\\",
  vars: {
    \\"tjd0u1n-0\\": [() => window.innerWidth * 0.5, \\"px\\"]
  }
});
export const T2 =
/*#__PURE__*/
styled(\\"h2\\")({
  name: \\"T2\\",
  class: \\"T2_t1jdyzzg\\",
  mod: {
    \\"t1jdyzzg__primary_0\\": p => p.primary$
  }
});
export const T3 =
/*#__PURE__*/
styled(\\"h3\\")({
  name: \\"T3\\",
  class: \\"T3_t1y4fq36\\"
});

// Injected by Linaria Loader for Webpack Compilation (virtual)
require(\\"./.linaria-cache/css/d8444e21c14a3c7e7c923bbcf86d4d84.dev.linaria.css\\");"
`;
